### PR TITLE
gh-78878: Fix crash when creating an instance of `_ctypes.CField`

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue34697

* Issue: gh-78878

### CODE REVIEW

**Comment by shihai1991** [comment]
@serhiy-storchaka @izbyshev  Hi, guys. I have try to solve this probleam, pls review this patch, thank you :)

**Review by MaxwellDupre — APPROVED** [review]
unittest Lib/ctypes/test/test_struct_fields.py
Ran 8 tests in 0.005s
OK
Looks ok.

**Comment by kumaraditya303** [comment]
@shihai1991 Can you fix the conflicts ? Thanks

**Comment by shihai1991** [comment]
> @shihai1991 Can you fix the conflicts ? Thanks

Of course. I forget detail already. I will take a look again.

**Review by iritkatriel — DISMISSED** [review]
This has merge conflicts now.

**Comment by shihai1991** [comment]
Resolve the conflict again :)

**Comment by shihai1991** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Core and Builtins/2022-12-18-19-07-29.gh-issue-78878.00QOp1.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove the ``PyCField_new`` which user cannot call it directly.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Core and Builtins/2022-12-18-19-07-29.gh-issue-78878.00QOp1.rst]:
```suggestion
Fix crash when creating an instance of :class:`!_ctypes.CField`.
```

Comment (kumaraditya303) [reply Misc/NEWS.d/next/Core and Builtins/2022-12-18-19-07-29.gh-issue-78878.00QOp1.rst]: Also the news entry should be in Library not Core and Builtins.

Comment (shihai1991) [reply Misc/NEWS.d/next/Core and Builtins/2022-12-18-19-07-29.gh-issue-78878.00QOp1.rst]: Updated, thanks.

**Review by kumaraditya303 — APPROVED** [review]
LGTM

**Comment by kumaraditya303** [comment]
Let's only backport to 3.11.

**Comment by shihai1991** [comment]
It's an very old MR. Thanks everyone's review. Thanks Kumar for your merge.

**Pull request merged**